### PR TITLE
RavenDB-17217 fix test and remove has revision flag from tombstone de…

### PR DESCRIPTION
**Pull request by efratshenhar**
…stination

### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-17217

### Additional description

Added wait for the removal of has revision flag after enforcing revision configuration.
remove has revision flag at destination (external replication) after enforcing revision configuration at the source.

_Please delete below the options that are not relevant_

### Type of change

- Bug fix

### How risky is the change?

- Low 

### Backward compatibility

- Not relevant

### Is it platform specific issue?

- No

### Documentation update

- No documentation update is needed 

### Testing 

- Tests have been added that prove the fix is effective or that the feature works

### Is there any existing behavior change of other features due to this change?

- No

### UI work

- No UI work is needed

### CODE REVIEW

File: src/Raven.Server/Documents/DocumentsStorage.cs [hunk 1]
```diff
@@ -1498,11 +1498,18 @@ public static Tombstone TableValueToTombstone(JsonOperationContext context, ref
                 else
                 {
                     flags = localFlags | documentFlags;
+                    var revisionsStorage = DocumentDatabase.DocumentsStorage.RevisionsStorage;
 
+                    if (nonPersistentFlags.Contain(NonPersistentDocumentFlags.FromReplication) &&
+                        revisionsStorage.Configuration == null &&
```
Comment (karmeli87) [inline src/Raven.Server/Documents/DocumentsStorage.cs]: If we have a local tombstone and we receive a tombstone which is a conflict resolution.. wouldn't we strip the flag here?

File: src/Raven.Server/Documents/DocumentsStorage.cs [hunk 2]
```diff
@@ -1498,11 +1498,18 @@ public static Tombstone TableValueToTombstone(JsonOperationContext context, ref
                 else
                 {
                     flags = localFlags | documentFlags;
+                    var revisionsStorage = DocumentDatabase.DocumentsStorage.RevisionsStorage;
 
+                    if (nonPersistentFlags.Contain(NonPersistentDocumentFlags.FromReplication) &&
+                        localFlags.Contain(DocumentFlags.HasRevisions) != documentFlags.Contain(DocumentFlags.HasRevisions))
+                    {
+                        var (_, count) = revisionsStorage.GetRevisions(context, id, 0, 1);
```
Comment (karmeli87) [inline src/Raven.Server/Documents/DocumentsStorage.cs]: We don't want that, it will materialize ~~all the revisions~~ a revision.

Comment (karmeli87) [reply src/Raven.Server/Documents/DocumentsStorage.cs]: maybe `GetRevisionsCount` ?

Comment (efratshenhar) [reply src/Raven.Server/Documents/DocumentsStorage.cs]: done

**Review by karmeli87 — APPROVED** [review]

**Pull request merged**